lib/jobs/local-catalog.js

```diff
@@ -0,0 +1,189 @@
+    new di.Inject(
+        'Job.Base',
+        'JobUtils.CommandParser',
+        'Services.Waterline',
+        'Logger',
+        'Promise',
+        'Assert',
+        'Util',
+        '_',
+        'Services.Encryption',
+        'ChildProcess'
+    )
+);
+function LocalCatalogJobFactory(
+    BaseJob, 
+    parser, 
+    waterline, 
+    Logger, 
+    Promise, 
+    assert, 
+    util, 
+    _,
+    encryption,
+    ChildProcess
+) {
+    var logger = Logger.initialize(LocalCatalogJobFactory);
+
+    /**
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function LocalCatalogJob(options, context, taskId) {
+        LocalCatalogJob.super_.call(this, 
+                                   logger,
+                                   options, 
+                                   context, 
+                                   taskId);
+
+        assert.string(this.context.target);
+        assert.arrayOfString(this.options.commands);
+    
+        this.childProcess = undefined;
+        this.nodeId = this.context.target;
+        this.commands = _.isArray(options.commands) ? options.commands : [options.commands];
+    }
+    util.inherits(LocalCatalogJob, BaseJob);
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype._run = function() {
+        var self = this;
+
+        waterline.nodes.findByIdentifier(self.nodeId)
+        .then(function (node) {
+            assert.ok(node, 'No node for local catalog');
+            return Promise.map(self.commands, function(cmd) {
+                return parser.validateParser(cmd)
+                .then(function() {
+                    return self.runCommand(cmd);
+                })
+                .catch(function(err) {
+                    logger.error("Run error", { command: cmd, error: err });
+                });
+            });
+        })
+        .then(self.handleResponse.bind(self))
+        .then(function() {
+            self.childProcess = undefined;
+            self._done();
+        })
+        .catch(function(err) {
+            self.childProcess = undefined;
+            self._done(err);
+        });
+    };
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype.runCommand = function (cmd) {
+        var self = this;
+        var args = cmd.split(' ');
+        var program = args[0];
+        if(program === "sudo") {
+            program = args[1];
+            args.shift();
+        }
+        args.shift();
+        return Promise.resolve()
+        .then(function() {
+            var env = {
+                'nodeid': self.nodeId
+            };
+            self.childProcess = new ChildProcess(program, args, env);
+
+            logger.debug("Sending command to node.", {
+                command: cmd,
+                nodeId: self.nodeId
+            });
+
+            return self.childProcess.run({ retries: 0, delay: 0 });
```

Comment:
@iceiilin  execFile() is encapsulated in on-core/lib/common/child-process.js and doesn't set the timeout option.   We can revisit this in another PR when timeout is supported there.